### PR TITLE
Fix external Intent problem. 

### LINKED ISSUE #477: External Intent problem.
When Google Plus link is opened in the app, it look like this. It is a famous Android problem. In order to solve it we need to add couple of flags into the Intent. 

In the below screenshot, you can see that Google+ app is open with red primaryColor but with our GDG app logo. 

screenshot_2015-08-30-23-44-32

### PR DESCRIPTION
**Pull request by tasomaniac**
Introduced new utility function `createExternalIntent`
It creates and Intent with ACTION_VIEW with appropriate flags and extras.

Fixes #477 

Old screenshot can be found in #477 
It now looks like this. GDG app and Google+ are separate entities now. 
image

### CODE REVIEW

File: app/src/main/java/org/gdg/frisbee/android/utils/Utils.java [hunk 1]
```diff
@@ -209,8 +212,18 @@ public static boolean isEmailAddress(String possibleEmail) {
...
 
-
     public static boolean canLaunch(Context context, final Intent viewUrlIntent) {
         return context.getPackageManager().resolveActivity(viewUrlIntent, PackageManager.MATCH_DEFAULT_ONLY) != null;
     }
+
+    @SuppressWarnings("deprecation")
+    public static Intent createExternalIntent(Context context, Uri uri) {
+        Intent intent = new Intent(Intent.ACTION_VIEW, uri);
+        intent.putExtra(ShareCompat.EXTRA_CALLING_PACKAGE, context.getPackageName());
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/utils/Utils.java]: This Extra's are taken from ShareCompat's implementation. The external Activity we open might use this information if they want to using ShareCompat.IntentReader.

File: app/src/main/java/org/gdg/frisbee/android/utils/Utils.java [hunk 2]
```diff
@@ -209,8 +212,18 @@ public static boolean isEmailAddress(String possibleEmail) {
...
+        if (context instanceof Activity) {
+            intent.putExtra(ShareCompat.EXTRA_CALLING_ACTIVITY, ((Activity) context).getComponentName());
+        }
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET);
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/utils/Utils.java]: The most important thing is this flag.

**Comment by friedger** [comment]
Would be nice to add a custom chrome cast tab. Maybe out of scope?

**Comment by tasomaniac** [comment]
Definitely. But that should be another issue. 
When opening a Google Plus link I don't think Custom tabs are necessary but for the other things like Help, Wisdom Book, custom tabs would be awesome.

**Pull request merged**